e2e/e2e_test.go

```diff
@@ -0,0 +1,172 @@
+	acc, _ := hedera.AccountIDFromString(configuration.Hedera.Client.Operator.AccountId)
+	receiving, _ := hedera.AccountIDFromString(configuration.Hedera.Watcher.CryptoTransfer.Accounts[0].Id)
+	topicID, _ := hedera.TopicIDFromString(configuration.Hedera.Watcher.ConsensusMessage.Topics[0].Id)
+	ethConfig := configuration.Hedera.Eth
+
+	accID, _ := hedera.AccountIDFromString(configuration.Hedera.Client.Operator.AccountId)
+	pK, _ := hedera.PrivateKeyFromString(configuration.Hedera.Client.Operator.PrivateKey)
+
+	client := initClient(accID, pK)
+	ethClient := ethclient.NewEthereumClient(ethConfig)
+	whbarInstance, err := whbar.NewWhbar(whbarContractAddress, ethClient.Client)
+
+	// Get the wrapped hbar balance of the receiver before the transfer
+	whbarBalanceBefore, err := whbarInstance.BalanceOf(&bind.CallOpts{}, whbarReceiverAddress)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	fmt.Printf("WHBAR balance before transaction: [%s]\n", whbarBalanceBefore)
+
+	// Get custodian hbar balance before transfer
+	recieverBalance, err := hedera.NewAccountBalanceQuery().
+		SetAccountID(receiving).
+		Execute(client)
+
+	fmt.Println(fmt.Sprintf(`HBAR custodian balance before transaction: [%d]`, recieverBalance.Hbars.AsTinybar()))
+
+	// Get the transaction receipt to verify the transaction was executed
+	transactionResponse, err := sendTransactionToCustodialAccount(acc, receiving, memo, hBarAmount, client)
+	transactionReceipt, err := transactionResponse.GetReceipt(client)
+
+	if err != nil {
+		fmt.Println(fmt.Sprintf(`Transaction unsuccessful, Error: [%s]`, err))
+		t.Fatal(err)
+	} else {
+		fmt.Println(fmt.Sprintf(`Successfully sent HBAR to custodian adress, Status: [%s]`, transactionReceipt.Status))
+	}
+
+	// Get custodian hbar balance after transfer
+	recieverBalanceNew, err := hedera.NewAccountBalanceQuery().
+		SetAccountID(receiving).
+		Execute(client)
+
+	fmt.Println(fmt.Sprintf(`HBAR custodian balance after transaction: [%d]`, recieverBalanceNew.Hbars.AsTinybar()))
+
+	// Verify that the custodial address has recieved exactly the amount sent
+	if (recieverBalanceNew.Hbars.AsTinybar() - recieverBalance.Hbars.AsTinybar()) != hedera.HbarFrom(hBarAmount, "hbar").AsTinybar() {
+		t.Fatal(`Expected to recieve the exact transfer amount of hbar`)
+	}
+
+	// Subscribe to Topic
+	hedera.NewTopicMessageQuery().
+		SetStartTime(time.Unix(0, time.Now().UnixNano())).
+		SetTopicID(topicID).
+		Subscribe(
+			client,
+			func(response hedera.TopicMessage) {
+				msg := &validatorproto.TopicSubmissionMessage{}
+				proto.Unmarshal(response.Contents, msg)
+
+				if msg.GetType() == validatorproto.TopicSubmissionType_EthSignature {
+					//Verify that all the submitted messages have signed the same transaction
+					if msg.GetTopicSignatureMessage().TransactionId != fromHederaTransactionID(transactionResponse.TransactionID) {
+						t.Fatal(`Expected signature message to contain the transaction id`)
```

Comment:
Probably we can add more information about transaction id with t.Fatalf()

Reply:
Yep, added more details as discussed